docs/design-contracts/design-contracts.md

```diff
@@ -0,0 +1,412 @@
+
+The `change` box is created to make the input and output sums of the transaction balanced.
+The transaction fee box is omited for simplicity, in practice, it should be added
+explicitly because in the case of more complex transactions (like Swap) there are many
+ways to pay transaction fee.
+
+##### CancelBuy, CancelSell Transactions
+
+At any time, the `buyer` can cancel the order by sending `CancelBuy` transaction. The
+transaction should satisfy the guarding `buyOrder` contract which protects the `bid` box.
+As you can see on the diagram, both the `Cancel` and the `Swap` transactions can spend the
+`bid` box. When a box have spending alternatives (or _spending path_) then each
+alternative should be identified by unique name prefixed with `!` (`!cancel` and `!swap`
+for the `bid` box). Each alternative have specific spending conditions. In our example,
+when the `bid` box is spend by the `Cancel` transaction the `?buyer` condition should be
+satisfied, which can be read as "the signature of the buyer should be presented in the
+transaction". Therefore, only buyer can cancel the buy order. This "signature" condition
+is only required for `!cancel` spending alternative and not required for `!swap`.
+
+##### SellOrder Transaction
+
+The `SellOrder` transaction is similar to the `BuyOrder` it has to do with tokens in
+addition to ERGs. The transaction spends `E: ERG` and `T: TID` tokens from seller's wallet
+(specified as `pk(seller)` contract). The two outputs are `ask` and `change`. The change
+is a standard box to balance transaction. The `ask` box stores `tAmt: TID` tokens to swap
+and the `minErg: ERG` - a minimum amount of ERGs required in every box.
+
+##### Swap Transaction
+
+This is a key transaction in the DEX scenario. The transaction has many of spending
+conditions on the input boxes and those conditions are included in the `buyOrder` and
+`sellOrder` contracts and consequently verified when the transaction is added to the
+blockchain. However, on the diagram those conditions are not specified in the `bid` and
+`ask` boxes, they are instead defined in the output boxes of the transaction. 
+
+This is because most of the conditions relate to the properties of the output boxes. This is
+usability convention, we could specify those properties in the `bid` box but, then we would
+had to use more complex expressions.
+
+Let's take the output created by the arrow labeled with `buyerOut@0`. This label tell us
+that the output is at the index `0` in the `OUTPUTS` collection of the transaction and that
+we can refer to this box by the `buyerOut` name. Thus we can label both the box itself and
+the arrow to give it a name.
+
+The conditions shown in the `buyerOut` box have the form `bid ? condition`, which means
+they should be checked before the `bid` box can be spent. 
+The conditions have the following meaning:
+1) `tAmt: TID` require the box to have `tAmt` amount of `TID` token
```

Comment:
@aslesarenko Initially I confuse `tAmt` from `bid ? tAmt: TID` with `tAmt` from `SellOrder`. I suggest exposing some contract internals (like token amount) inside the contract box on the diagram and give them different names.

Reply:
I should probably make is more clear. tAmt is not a field, it is a parameter of the diagram - Long value of tokens. Or put it another way, it is a shared parameter both for Sell Order and Swap transactions. So basically same name means the same value throughout the diagram. However, only explicit conditions marked with `?` are transformed to ErgoScript. But all others are actually ensured during transaction building (Appkit code)